src/runtime_src/core/common/api/xrt_device.cpp

```diff
@@ -396,6 +396,87 @@ open_context(xrt::aie::device::access_mode am)
   core_device->open_aie_context(am);
 }
 
+static void
+get_abs_col(const xrt_core::device* device, uint16_t context_id, uint16_t& col)
+{
+  auto data = xrt_core::device_query_default<xrt_core::query::aie_partition_info>(device, {});
+  for (const auto& entry : data) {
+    if (std::stoi(entry.metadata.id) == context_id) {
+      col += entry.start_col;
+      if (col >= entry.num_cols)
+        throw std::out_of_range("col index out of range");
+      return;
+    }
```

Comment:
While I am okay with this as is (it reads well), you may want to reverse/continue and use early-exit thereby saving one level of indentation.